src/Castle.Core.Tests/DynamicProxy.Tests/SerializableClassTestCase.cs

```diff
@@ -250,8 +251,8 @@ public void ProxyGenerationOptionsRespectedOnDeserialization()
 			Assert.AreEqual(proxy.GetType(), proxy.GetType().GetMethod("get_Current").DeclaringType);
 			Assert.AreNotEqual(proxy.GetType(), proxy.GetType().GetMethod("CalculateSumDistanceNow").DeclaringType);
 			Assert.AreEqual(proxy.GetType().BaseType, proxy.GetType().GetMethod("CalculateSumDistanceNow").DeclaringType);
-			var options2 =
-				(ProxyGenerationOptions)proxy.GetType().GetField("proxyGenerationOptions").GetValue(null);
+			var options2 = (ProxyGenerationOptions)proxy.GetType().
+				GetField("proxyGenerationOptions", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null);
```

Comment:
Why is this static a thing? Mostly instance based or have I missed something.

Reply:
`proxyGenerationOptions` is a static field (see BaseProxyGenerator.cs). If you don't pass any binding flags then you get `Instance|Static|Public` (see [Type.GetField()](https://referencesource.microsoft.com/#mscorlib/system/type.cs,31ecddf2e1a221ae)), so I need to set the flags manually to get it to return the private field.